File: 0622. Design Circular Queue/Solution.py
class MyCircularQueue:

    def __init__(self, k: int):
        self.k = k
        self.element = []
        

    def enQueue(self, value: int) -> bool:
        if len(self.element) < self.k:
            self.element.append(value)
            return True
        else:
            return False

    def deQueue(self) -> bool:
        try:
            self.element.pop(0)
            return True
        except:
            return False    
        

    def Front(self) -> int:
        try:
            return self.element[0]
        except:
            return -1
        

    def Rear(self) -> int:
        try:
            return self.element[-1]
        except:
            return -1
        

    def isEmpty(self) -> bool:
        return self.element == []
        

    def isFull(self) -> bool:
        return len(self.element) == self.k
        


# Your MyCircularQueue object will be instantiated and called as such:
# obj = MyCircularQueue(k)
# param_1 = obj.enQueue(value)
# param_2 = obj.deQueue()
# param_3 = obj.Front()
# param_4 = obj.Rear()
# param_5 = obj.isEmpty()
# param_6 = obj.isFull()

File: 0622. Design Circular Queue/Solution.cpp
class MyCircularQueue {
public:
    int k;
    std::vector<int> elements;
    
    MyCircularQueue(int k) {
        this-> k = k;
        this->elements.reserve(k);
    }
    
    bool enQueue(int value) {
        if (this->elements.size() < k)
        {
            this->elements.push_back(value);
            return true;
        }
        else
        {
            return false;
        }
    }
    
    bool deQueue() {
        if(this->elements.size() > 0)
        {
            this->elements.erase(this->elements.begin());
            return true;
        }
        else
        {
            return false;
        }
    }
    
    int Front() {
        if(this->elements.size() > 0)
        {
            return this->elements[0];
        }
        else
        {
            return -1;
        }
    }
    
    int Rear() {
        if(this->elements.size() > 0)
        {
            return this->elements[this->elements.size()-1];
        }
        else
        {
            return -1;
        }        
    }
    
    bool isEmpty() {
        return this->elements.size() == 0;
        
    }
    
    bool isFull() {
        return this->elements.size() == this->k;
    }
};

/**
 * Your MyCircularQueue object will be instantiated and called as such:
 * MyCircularQueue* obj = new MyCircularQueue(k);
 * bool param_1 = obj->enQueue(value);
 * bool param_2 = obj->deQueue();
 * int param_3 = obj->Front();
 * int param_4 = obj->Rear();
 * bool param_5 = obj->isEmpty();
 * bool param_6 = obj->isFull();
 */
